drake/core/Core.h

```diff
@@ -70,25 +130,81 @@ namespace Drake {
     };
     std::size_t size() { return vec1.size()+vec2.size(); }
 
+    friend Eigen::Matrix<ScalarType,Vector1<ScalarType>::RowsAtCompileTime + Vector2<ScalarType>::RowsAtCompileTime,1> toEigen(const CombinedVector<ScalarType,Vector1,Vector2>& vec) {
+      Eigen::Matrix<ScalarType,Vector1<ScalarType>::RowsAtCompileTime + Vector2<ScalarType>::RowsAtCompileTime,1> x;
+      x << toEigen(vec.vec1), toEigen(vec.vec2);
+      return x;
+    }
+
   private:
     Vector1<ScalarType> vec1;
     Vector2<ScalarType> vec2;
   };
 
-  template <template <typename> class Vector1, template <typename> class Vector2>
+  /** CombinedVectorBuilder<Vector1,Vector2>::type<ScalarType>
+   * @brief Provides an alias to use the CombinedVector class as a model of the Vector concept
+   * @concept{vector_concept}
+   *
+   * Uses template aliasing so that combining a vector with the NullVector simply returns the orginal vector type.
+   */
+
+  template <template <typename> class Vector1, template <typename> class Vector2, typename Enable1 = void, typename Enable2 = void>
```

Comment:
The user never has to think about it.  It just provides the logic to return the original type if you are combining with the NullVector.  That requires two enables to handle all four cases.  
That logic is important to have so that you can, for instance, pass in a PendulumState to simulate a system which is the Pendulum cascaded with the BotVisualizer.